lib/engine/game/g_18_zoo/step/choose_ability_on_or.rb

```diff
@@ -0,0 +1,261 @@
+          # process_choose_moles?(action) if action.choice['type'] == 'moles'
+          # process_choose_ancient_maps?(action) if action.choice['type'] == 'ancient_maps'
+          # process_choose_hole?(action) if action.choice['type'] == 'hole'
+          # process_choose_on_diet?(action) if action.choice['type'] == 'on_diet'
+          # process_choose_is_mine(action) if action.choice['type'] == 'that_is_mine'
+          # process_choose_work_in_progress?(action) if action.choice['type'] == 'work_in_progress'
+          # process_choose_two_barrels?(action) if action.choice['type'] == 'two_barrels'
+          # process_choose_bandage?(action) if action.choice['type'] == 'bandage'
+          # process_choose_wings?(action) if action.choice['type'] == 'wings'
+          # process_choose_sugar(action) if action.choice['type'] == 'sugar'
+        end
+
+        private
+
+        # # TODO: add additional logic
+        # def can_choose_rabbits?(entity)
+        #   @game.rabbits.owner == entity
+        # end
+
+        # # TODO: add additional logic
+        # def can_choose_moles?(entity)
+        #   @game.moles.owner == entity
+        # end
+
+        # # TODO: add additional logic
+        # def can_choose_ancient_maps?(entity)
+        #   @game.ancient_maps.owner == entity
+        # end
+
+        # TODO: add additional logic
+        # def can_choose_hole?(entity)
+        #   @game.hole.owner == entity
+        # end
+
+        # TODO: add additional logic
+        # def can_choose_on_diet?(entity)
+        #   @game.on_diet.owner == entity
+        # end
+
+        # TODO: add additional logic
+        # def can_choose_is_mine?(entity)
+        #   @game.that_is_mine.owner == entity
+        # end
+
+        # TODO: add additional logic
+        # def can_choose_work_in_progress?(entity)
+        #   @game.work_in_progress.owner == entity
+        # end
+
+        # TODO: add additional logic
+        # def can_choose_corn?(entity)
+        #   @game.corn.owner == entity
+        # end
+
+        # TODO: add logic
+        # def can_choose_two_barrels?(entity)
+        #   @game.two_barrels.owner == entity && !@game.two_barrels_used_this_or?(entity)
+        # end
+
+        # TODO: add additional logic
+        # def can_choose_bandage?(entity)
+        #   @game.bandage.owner == entity
+        # end
+
+        # TODO: add additional logic
+        # def can_choose_wings?(entity)
+        #   @game.wings.owner == entity
+        # end
+
+        # TODO: add additional logic
+        # def can_choose_sugar?(corporation)
+        #   @game.a_spoonful_of_sugar.owner == corporation &&
+        #     corporation.trains.any? { |train| !%w[2J 4J].include? train.name } &&
+        #     corporation.all_abilities.none? { |a| a.type == :increase_distance_for_train }
+        # end
+
+        def choices_for_zoo_ticket(company)
+          corporation = @game.current_entity
+          name = company.owner.name
```

Comment:
Do you need to verify that the corporation is owned by the same person as the company?

Reply:
done in 2edf5ad23